src/main/java/io/cdap/plugin/facebook/source/common/AdsInsightsTransformer.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.facebook.source.common;
+
+import com.facebook.ads.sdk.AdsInsights;
+import com.google.gson.JsonObject;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+/**
+ * This is helper class for transforming {@link AdsInsights} instance to {@link StructuredRecord}.
+ */
+public class AdsInsightsTransformer {
+  /**
+   * Transforms {@link AdsInsights} instance to {@link StructuredRecord} instance accordingly to given schema.
+   */
+  public static StructuredRecord transform(AdsInsights insights, Schema schema) {
+    StructuredRecord.Builder builder = StructuredRecord.builder(schema);
+    JsonObject insightsJson = insights.getRawResponseAsJsonObject();
+    insightsJson.entrySet().forEach(entry -> {
+      if (schemaContainsField(schema, entry.getKey())) {
+        Schema fieldSchema = schema.getField(entry.getKey()).getSchema();
+
+        if (fieldSchema.isNullable()) {
+          fieldSchema = fieldSchema.getNonNullable();
+        }
+
+        switch (fieldSchema.getType()) {
```

Comment:
why other types are not supported like integer, double, date, timestamp etc

Reply:
We are not sure why the original developers not support those types. Let me know if you want me to add a TO DO here. Facebook Ads plugins needs a lot of changes and is not very useful in the correct form. This feedback was provided to Bhooshan from our testing also also from P&G testing.

Reply:
Please add TODO with corresponding jira

Reply:
Code Updated with TODO and created JIRA ticket - https://issues.cask.co/browse/PLUGIN-365